modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -1267,6 +1910,322 @@ private String decryptKeyName(byte[] data) {
         });
     }
 
+    /**
+     * Two phase distributed process, that performs cache group encryption key rotation.
+     */
+    private class GroupKeyChangeProcess {
```

Comment:
Move to external class, e.g. with package visibility. GridEncryptionManager more 1000 lines are to match.